index.ts

```diff
@@ -0,0 +1,245 @@
+const EthQuery = require('ethjs-query')
+import assert from 'assert';
+const Mutex = require('await-semaphore').Mutex
+
+/**
+    @property opts.provider - A ethereum provider
+    @property opts.blockTracker - An instance of eth-blocktracker
+    @property opts.getPendingTransactions - A function that returns an array of txMeta
+    whosee status is `submitted`
+    @property opts.getConfirmedTransactions - A function that returns an array of txMeta
+    whose status is `confirmed`
+*/
+export interface opts {
+  provider: object;
+  blockTracker: any;
+  getPendingTransactions(address: string):Transactions[];
+  getConfirmedTransactions(address: string): Transactions[];
+}
+
+/**
+  * @property  highestLocallyConfirmed - A hex string of the highest nonce on a confirmed transaction.
+  * @property  nextNetworkNonce - The next nonce suggested by the eth_getTransactionCount method.
+  * @property  highestSuggested - The maximum between the other two, the number returned.
+  * @property  local - Nonce details derived from pending transactions and highestSuggested
+  * @property  network - Nonce details from the eth_getTransactionCount method
+*/
+export interface NonceDetails {
+  params:{
+  highestLocallyConfirmed: number;
+  nextNetworkNonce: number;
+  highestSuggested: number;
+  };
+  local: HighestContinuousFrom;
+  network: NetWorkNextNonce;
+}
+
+/**
+  * @property  nextNonce - The highest of the nonce values derived based on confirmed and pending transactions and eth_getTransactionCount method
+  * @property  nonceDetails - details of nonce value derivation.
+  * @property  releaseLock 
+*/
+export interface NonceLock{ 
+  nextNonce: number; 
+  nonceDetails: NonceDetails; 
+  releaseLock: any;
```

Comment:
This is a function, per our usage and the dependency's documentation: https://www.npmjs.com/package/await-semaphore
```suggestion
  releaseLock: () => void;
```